test/tests/serial/mail_with_ldap.bats

```diff
@@ -1,245 +1,314 @@
-
-  wait_for_smtp_port_in_container mail_with_ldap
+    dms-openldap
+
+  #
+  # Setup DMS container
+  #
+
+  local ENV_LDAP_CONFIG=(
+    # Configure for LDAP account provisioner and alternative to Dovecot SASL:
+    --env ACCOUNT_PROVISIONER=LDAP
+    --env ENABLE_SASLAUTHD=1
+    --env SASLAUTHD_MECHANISMS=ldap
+
+    # ENV to configure LDAP configs for Dovecot + Postfix:
+    # NOTE: `scripts/startup/setup.d/ldap.sh:_setup_ldap()` uses `_replace_by_env_in_file()` to configure settings (stripping `DOVECOT_` / `LDAP_` prefixes):
+    # Dovecot:
+    --env DOVECOT_PASS_FILTER='(&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%n))'
+    --env DOVECOT_TLS=no
+    --env DOVECOT_USER_FILTER='(&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%n))'
+    # Postfix:
+    --env LDAP_BIND_DN='cn=admin,dc=localhost,dc=localdomain'
+    --env LDAP_BIND_PW='admin'
+    --env LDAP_QUERY_FILTER_ALIAS='(|(&(mailAlias=%s)(objectClass=PostfixBookMailForward))(&(mailAlias=%s)(objectClass=PostfixBookMailAccount)(mailEnabled=TRUE)))'
+    --env LDAP_QUERY_FILTER_DOMAIN='(|(&(mail=*@%s)(objectClass=PostfixBookMailAccount)(mailEnabled=TRUE))(&(mailGroupMember=*@%s)(objectClass=PostfixBookMailAccount)(mailEnabled=TRUE))(&(mailalias=*@%s)(objectClass=PostfixBookMailForward)))'
+    --env LDAP_QUERY_FILTER_GROUP='(&(mailGroupMember=%s)(mailEnabled=TRUE))'
+    --env LDAP_QUERY_FILTER_SENDERS='(|(&(mail=%s)(mailEnabled=TRUE))(&(mailGroupMember=%s)(mailEnabled=TRUE))(|(&(mailAlias=%s)(objectClass=PostfixBookMailForward))(&(mailAlias=%s)(objectClass=PostfixBookMailAccount)(mailEnabled=TRUE)))(uniqueIdentifier=some.user.id))'
+    --env LDAP_QUERY_FILTER_USER='(&(mail=%s)(mailEnabled=TRUE))'
+    --env LDAP_SEARCH_BASE='ou=people,dc=localhost,dc=localdomain'
+    --env LDAP_SERVER_HOST="${FQDN_LDAP}"
+    --env LDAP_START_TLS=no
+  )
+
+  # Extra ENV needed to support specific testcases:
+  local ENV_SUPPORT=(
+    --env PERMIT_DOCKER=container # Required for attempting SMTP auth on port 25 via nc
+    # Required for openssl commands to be successul:
+    # NOTE: snakeoil cert is created (for `docker-mailserver.invalid`) via Debian post-install script for Postfix package.
+    # TODO: Use proper TLS cert
+    --env SSL_TYPE='snakeoil'
+
+    # TODO; All below are questionable value to LDAP tests?
+    --env POSTMASTER_ADDRESS="postmaster@${FQDN_LOCALHOST_A}" # TODO: Only required because LDAP accounts use unrelated domain part. FQDN_LOCALHOST_A / ldif files can be adjusted to FQDN_MAIL
+    --env PFLOGSUMM_TRIGGER=logrotate
+    --env REPORT_RECIPIENT=1 # TODO: Invalid value, should be a recipient address (if not default postmaster), remove?
+    --env SPOOF_PROTECTION=1
+  )
+
+  local CUSTOM_SETUP_ARGUMENTS=(
+    --hostname "${FQDN_MAIL}"
+    --network "${DMS_TEST_NETWORK}"
+
+    "${ENV_LDAP_CONFIG[@]}"
+    "${ENV_SUPPORT[@]}"
+  )
+
+  # Set default implicit container fallback for helpers:
+  export CONTAINER_NAME
+  CONTAINER_NAME=${CONTAINER1_NAME}
```

Comment:
Should be fine to just assign immediately, right?

```suggestion
  export CONTAINER_NAME=${CONTAINER1_NAME}
```

Reply:
Yep, initially I thought I was going to need it for the other container 😅 

There's the other LDAP PR open at present that does require introducing some additional DMS variants to test an ENV though.